third_party/amd/lib/TritonAMDGPUToLLVM/DotOpToLLVM/MFMA.cpp

```diff
@@ -263,6 +264,13 @@ struct DotOpMFMAConversionHelper {
     Type structTy = LLVM::LLVMStructType::getLiteral(
         ctx, SmallVector<Type>(fc.size(), dstElemTy));
     Value res = packLLElements(loc, typeConverter, fc, rewriter, structTy);
+
+    op->getBlock()->walk([&](amdgpu::InstructionSchedHint schedHint) {
```

Comment:
> so that it's a bit better isolation and aggregating the logic all in `SchedInstruction`.

Agree

> the current way of getting `InstructionSchedHint` works if we only have one dot and on hint op in the block.

Yes

> It will fall apart for cases like attention. We need to refine the attaching logic later by going through some IR.

Agree. It will fall apart. I would suggest to move forward with instruction scheduling for `tt.dot` and come back later to the attention kernel.